jax/_src/numpy/lax_numpy.py

```diff
@@ -6702,6 +6702,12 @@ def trace(a: ArrayLike, offset: int | ArrayLike = 0, axis1: int = 0, axis2: int
   util.check_arraylike("trace", a)
   if out is not None:
     raise NotImplementedError("The 'out' argument to jnp.trace is not supported.")
+
+  axis1 = _canonicalize_axis(axis1, a.ndim)
+  axis2 = _canonicalize_axis(axis2, a.ndim)
+  if axis1 == axis2:
+    raise ValueError(f"axis1 and axis2 can not be same. axis1={axis1} and axis2={axis2}")
```

Comment:
Small nitpick: the error message here now prints the canonicalized version of the axis rather than the actual value passed by the user. This may be misleading to any user that hits this error.

Reply:
Thanks! Updated the code so that the error message now prints the actual values of axis1 and axis2  passed by the user.